lib/Textile/index.ts

```diff
@@ -223,36 +223,30 @@ class Textile extends API {
   }
 
   // Client should use this once account is onboarded to register with Cafe
-  getCafeSessions = async (): Promise<ReadonlyArray<ICafeSession>> => {
+  getCafeSessions = async (): Promise<ReadonlyArray<pb.CafeSession.AsObject>> => {
     const sessions = await this.cafeSessions()
     if (!sessions) {
       return []
     }
-    const values: ReadonlyArray<ICafeSession> | undefined | null = sessions.values
-    if (!values) {
-      return []
-    } else {
-      return values
-    }
+    return sessions.valuesList
   }
 
   // Client should use this if cafe sessions are detected as expired
-  getRefreshedCafeSessions = async (): Promise<ReadonlyArray<ICafeSession>> => {
+  getRefreshedCafeSessions = async (): Promise<ReadonlyArray<pb.CafeSession.AsObject>> => {
     const sessions = await this.cafeSessions()
     if (!sessions) {
       return []
     }
-    const values: ReadonlyArray<ICafeSession> | undefined | null = sessions.values
-    if (!values) {
-      return []
-    } else {
-      const refreshedValues: ReadonlyArray<ICafeSession> = await Promise.all(
-        values
-          .map(async (session) => await this.refreshCafeSession(session.id!))
-          .filter((session) => session !== undefined) as ReadonlyArray<ICafeSession>
-      )
-      return refreshedValues
-    }
+    const refreshedValues = await Promise.all(
+      sessions.valuesList.map(async (session) => await this.refreshCafeSession(session.id))
+    )
+    const reduced = refreshedValues.reduce<pb.CafeSession.AsObject[]>((acc, val) => {
```

Comment:
Here's where i'm now filtering out `undefined` results of the promises using `reduce`